test-utils/src/helpers.rs

```diff
@@ -73,14 +73,13 @@ where
 	CommunityIdentifier::new(location, bs).unwrap()
 }
 
-pub fn assert_last_event<T: frame_system::Config>(
-	generic_event: <T as frame_system::Config>::Event,
-) {
+pub fn last_event<T: frame_system::Config>() -> Option<T::Event> {
```

Comment:
I believe assertions should only be made within the test routine, otherwise it will not be clear what caused the assertion to  fail

Reply:
I tend to disagree;

* When it fails, one needs to investigate properly anyhow. So I prefer readability over the exact line of code where it fails.
* Also, upstream changes need to be incorporated, which means now we might have to update every event assertion whereas otherwise adjusting the function might have sufficed.

However, no strong opinion here...

Reply:
The priblem is that you don't get a trace. If you call that fn twice in one test, you don't easily find out if the error occured the first or the second time